packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -21,44 +19,79 @@ const _runInProfileModeDocsUrl =
 const _profileGranularityDocsUrl =
     'https://flutter.dev/docs/development/tools/devtools/performance#profile-granularity';
 
-class BannerMessagesController implements DisposableController {
-  final _dismissedMessageIds = <String>{};
+class BannerMessagesController {
+  final _messages = <DevToolsScreenType, List<BannerMessage>>{};
```

Comment:
make the values in this map
ValueNotifier<List<BannerMessage>>
and be sure to call 
notifyListeners on the ValueNotifier every time you change the content of the list.
Then users of this controller can be notified when the controller state changes instead of having to hope they call setState at the right time.

Reply:
the reason I didn't do this originally is because notifyListeners is protected
![Screen Shot 2020-03-27 at 7 28 22 AM](https://user-images.githubusercontent.com/43759233/77766235-9924f800-6ffc-11ea-9cd6-73778cdb66d3.png)

Reply:
sounds good. We could add a TODO to create our own sublcass that has a method to add an element to an existing List and call notifyListeners.